server/openapi_server/controllers/teacher_controller.py

```
@@ -0,0 +1,79 @@
+
+@teachers_router.post("/", response_model=Teacher)
+def create_teacher(teacher: Teacher):
+    """Create a new teacher."""
+    try:
+        result = save_teacher_to_mongo('teachers', teacher)
+        if result.get("acknowledged"):
+            teacher.id = str(result.get("inserted_id"))  # Set the generated ID in the Teacher object
+            return teacher
+        raise HTTPException(status_code=500, detail="Failed to create teacher")
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)